kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/batch/TableWriter.java

```diff
@@ -141,34 +144,47 @@ public String getTopic() {
     return topic;
   }
 
-  public static class Builder {
+  public static class Builder implements TableWriterBuilder {
     private final BigQueryWriter writer;
     private final PartitionedTableId table;
     private final String topic;
 
     private List<RowToInsert> rows;
 
+    private RecordConverter<Map<String, Object>> recordConverter;
+
     /**
      * @param writer the BigQueryWriter to use
      * @param table the BigQuery table to write to.
      * @param topic the kafka source topic associated with the given table.
      */
-    public Builder(BigQueryWriter writer, PartitionedTableId table, String topic) {
+    public Builder(BigQueryWriter writer, PartitionedTableId table, String topic,
+                   RecordConverter<Map<String, Object>> recordConverter) {
       this.writer = writer;
       this.table = table;
       this.topic = topic;
 
       this.rows = new ArrayList<>();
+
+      this.recordConverter = recordConverter;
     }
 
     /**
-     * Add a row to the builder.
-     * @param rowToInsert the rows to add.
+     * Add a record to the builder.
+     * @param recordToInsert the record to add.
      */
-    public void addRow(RowToInsert rowToInsert) {
+    public void addRecord(SinkRecord recordToInsert) {
```

Comment:
ah, I was definetly thinking that addRecord wouldn't take a SinkRecord, but instead the Map<String, Object> returned from the recordConverter. That way each Builder doesn't need to so the same recordConversion work.

Reply:
After a bit of conversation on this, I've changed my mind. Lets pass in SinkRecord here and the recordConverter to the constructor.